apps/list.c

```diff
@@ -945,6 +945,44 @@ static void list_options_for_command(const char *command)
     BIO_printf(bio_out, "- -\n");
 }
 
+static int is_md_available(const char *name)
+{
+    EVP_MD *md;
+
+/* Look through providers' digests*/
+    ERR_set_mark();
+    md = EVP_MD_fetch(NULL, name, NULL);
+    ERR_pop_to_mark();
+    if (md != NULL) {
+        EVP_MD_free(md);
+        return 1;
+    }
+
+#ifndef OPENSSL_NO_DEPRECATED_3_0
```

Comment:
I don't think this guard is necessary. Won't `get_digest_from_engine()` just return NULL in a no-deprecated build?

Reply:
I can move this #ifdef to the function, if it makes the code clearer

Reply:
I think it can be removed entirely. If you're in a no-deprecated build, then you have no-engine implicitly.

Reply:
Fixed. Thanks!